modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1363,7 +1363,7 @@ void ONNXImporter::parseSplit(LayerParams& layerParams, const opencv_onnx::NodeP
     {
         DictValue splits = layerParams.get("split");
         const int numSplits = splits.size();
-        CV_Assert(numSplits > 1);
+        CV_Assert(numSplits >= 1);
 
         std::vector<int> slicePoints(numSplits - 1, splits.get<int>(0));
```

Comment:
How to deal with a zero size vector?

Reply:
Not sure about zero size vectors. But when assertion is  such that `numSplits > 1` OpenCV parser does not pass this construction because of accretion: 

```python 

data  = [torch.ones(N, C, W, H),]

for x in data: 
    x = some processing

```

Reply:
I mean the code below which is incorrect for empty vector

Reply:
So we should not get pointer of the empty vector:
```cpp
std::vector<int> slicePoints(numSplits - 1, splits.get<int>(0));
layerParams.set("slice_point", DictValue::arrayInt(&slicePoints[0], slicePoints.size()));
```
Can you please create a test? Even without ONNX, just a Split layer initialized manually. See https://github.com/opencv/opencv/blob/4.x/modules/dnn/test/test_layers.cpp for examples.

Reply:
Added test with split = 1

Reply:
All tests seem to be passing. Added test generator is [here](https://github.com/opencv/opencv_extra/pull/1067)

Reply:
I'd like suggest the following approach. Having a zero size vector and accessing it's memory is not safe:
```cpp
if (numSplits == 1)
{
    layerParams.set("num_split", 1);
}
else
{
    CV_Assert(numSplits > 1);

    std::vector<int> slicePoints(numSplits - 1, splits.get<int>(0));
    for (int i = 1; i < splits.size() - 1; ++i)
    {
        slicePoints[i] = slicePoints[i - 1] + splits.get<int>(i);
    }
    layerParams.set("slice_point", DictValue::arrayInt(&slicePoints[0], slicePoints.size()));
}
```